### PR TITLE
fix: Primary color, error color

### PR DESCRIPTION
**Pull request by Vl4dk0**

### CODE REVIEW

File: out.py [hunk 1]
```diff
@@ -15,14 +15,15 @@ class _CursorInfo(ctypes.Structure):
 
 
 def error(message: str, context: Any = __UNIT) -> None:
-    print(message, file=sys.stderr)
+    print(ERROR + message + RESET, file=sys.stderr)
```
Comment (Hackder) [inline out.py:18]: print the error output using the helper out.println function
adding an optional `file` parameter might be required.

File: out.py [hunk 2]
```diff
@@ -15,14 +15,15 @@ class _CursorInfo(ctypes.Structure):
...
     if context is not __UNIT:
         print(context, file=sys.stderr)
 
 
 RESET = "\033[0m"
-PRIMARY = "\033[95m"
+PRIMARY = "\033[92m"
 SECONDARY = "\033[94m"
+ERROR = "\033[91m"
```
Comment (Hackder) [inline out.py:26]: Expose this color through a similar helper as the other ones.
Renaming it to DANGER might be required due to the conflict with the error printing `error` function

**Review by Hackder — CHANGES_REQUESTED** [review]

File: list_api/mock_list_api_failure.py
Comment (Hackder) [inline list_api/mock_list_api_failure.py:1]: What is the point of this mock?
If every request fails, you can't get past the login. Ideally you would want to be able to fail a specific request. Not just a request type, but a request. e.g. the second solution submit...

Modifying the existing mock and adding a raise seems good enough for now and much less complex

We can revisit this if we introduce automated tests somewhere in the future

File: list_api/mock_list_api_failure.py [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+
+from .models import Course, Problem, Submit, ListSession
+
+chance_to_success = 0.0
+
+
+def login(email: str, password: str) -> ListSession:
+    time.sleep(1)
+    if random.random() > chance_to_success:
+        out.error("Login failed, bad server status code: 500", "Login failed")
```
Comment (Hackder) [inline list_api/mock_list_api_failure.py:14]: the context paramter is for some "contextual" information, no for the same thing as the message. You don't have to use it all the time. It is for something like the causing exception etc.

File: out.py [hunk 4]
```diff
@@ -15,18 +15,19 @@ class _CursorInfo(ctypes.Structure):
 
 
 def error(message: str, context: Any = __UNIT) -> None:
-    print(message, file=sys.stderr)
+    println(danger(message), file=sys.stderr)
     if context is not __UNIT:
         print(context, file=sys.stderr)
```
Comment (Hackder) [inline out.py:20]: print context using the println function as well

File: out.py [hunk 5]
```diff
@@ -15,18 +15,19 @@ class _CursorInfo(ctypes.Structure):
...
 RESET = "\033[0m"
-PRIMARY = "\033[95m"
+PRIMARY = "\033[92m"
 SECONDARY = "\033[94m"
+DANGER = "\033[91m"
 
 
-def println(*args, end: str = "\n") -> None:
-    sys.stdout.write(" ".join(map(str, args)) + end)
+def println(*args, end: str = "\n", file: Any = sys.stdout) -> None:
```
Comment (Hackder) [inline out.py:29]: The type of the filestream is `TextIO`

File: out.py [hunk 6]
```diff
@@ -15,18 +15,19 @@ class _CursorInfo(ctypes.Structure):
...
+    print(" ".join(map(str, args)), end=end, file=file)
```
Comment (Hackder) [inline out.py:30]: If you use the built int print function, `...join(map..` is not required. You can just pass the args directly

File: out.py [hunk 7]
```diff
@@ -15,18 +15,19 @@ class _CursorInfo(ctypes.Structure):
...
 
 
 def flush() -> None:
```
Comment (Hackder) [inline out.py:33]: Flush should also have an optional `file` argument

**Review by Hackder — CHANGES_REQUESTED** [review]

File: out.py (hunk 5 above)
Comment (Vl4dk0) [reply out.py:29]: by Any I meant the possibility to also pass  string for exact filenames, i.e: "logs.txt".

File: list_api/mock_list_api_failure.py
Comment (Vl4dk0) [reply list_api/mock_list_api_failure.py:1]: should I just remove the file completely ?

File: out.py (hunk 5 above)
Comment (Hackder) [reply out.py:29]: That won't work, that will throw an error. You cannot pass a string into the `file` kwarg in the print function. If you look at the definition of `print` you will see it also uses `TextIO` as the type for the file kwarg.

File: out.py (hunk 7 above)
Comment (Vl4dk0) [reply out.py:33]: You should elaborate.
I dont understand what for. 
Like to only flush some specific TextIO file ?

File: out.py (hunk 5 above)
Comment (Hackder) [reply out.py:29]: Also, we are building a CLI application, that means that logging and printing to the screen are two different things for out. The out module is for printing things the user will see. We can use the built it python `logging` module for debug or application logs

File: list_api/mock_list_api_failure.py
Comment (Hackder) [reply list_api/mock_list_api_failure.py:1]: yes

File: out.py (hunk 7 above)
Comment (Hackder) [reply out.py:33]: When you flush, you flush a specific stream
`sys.stdout.flush()` will flush only the stdout. That means if you print something to `sys.stderr` it wouldn't get flushed by the call to the `out.flush()` function.

File: config-template.toml
Comment (Hackder) [inline config-template.toml:1]: These changes don't belong in a PR for a color change

File: list_api/api.py
Comment (Hackder) [inline list_api/api.py:1]: remove the failure if-check and revert back to "MOCK_LIST" for shorter env

File: out.py [hunk 8]
```diff
@@ -15,22 +15,27 @@ class _CursorInfo(ctypes.Structure):
...
 
-def println(*args, end: str = "\n") -> None:
-    sys.stdout.write(" ".join(map(str, args)) + end)
+def println(*args, end: str = "\n", file: TextIO = sys.stdout) -> None:
+    print(*args, end=end, file=file)
 
 
-def flush() -> None:
-    sys.stdout.flush()
+def flush(file: TextIO = None) -> None:
```
Comment (Hackder) [inline out.py:33]: inconsistent behavior with out.println

**Review by Hackder — CHANGES_REQUESTED** [review]

File: out.py (hunk 8 above)
Comment (Vl4dk0) [reply out.py:33]: what do you mean ?

Comment (Hackder) [reply out.py:33]: println with no arguments prints only to stdout, therefore flush with no arguments should flush only stdout

**Review by Hackder — APPROVED** [review]

**Pull request merged**